### ISSUE TITLE
[FranceTVSite] [FranceTV] No video formats found! (HTTP Error 400: Bad Request) france.tv

### ISSUE THREAD

**Issue by Gigoince** [yt-dlp/yt-dlp#9323]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

France

### Provide a description that is worded well enough to be understood

Downloading videos from france.tv is not possible anymore (as of 29/02/2024 - 10:00 UTC) 

Unable to download JSON metadata: HTTP Error 400: Bad Request
ERROR: 'NoneType' object has no attribute 'strip'

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
laurent@inoculum:Vidéos$ yt-dlp -vU -F https://www.france.tv/france-2/laissez-vous-guider/laissez-vous-guider-saison-1/5706441-les-merveilles-de-l-egypte-antique.html
[debug] Command-line config: ['-vU', '-F', 'https://www.france.tv/france-2/laissez-vous-guider/laissez-vous-guider-saison-1/5706441-les-merveilles-de-l-egypte-antique.html']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.12.30 from yt-dlp/yt-dlp [f10589e34] (zip)
[debug] Python 3.11.2 (CPython x86_64 64bit) - Linux-6.1.0-17-amd64-x86_64-with-glibc2.36 (OpenSSL 3.0.11 19 Sep 2023, glibc 2.36)
[debug] exe versions: ffmpeg 5.1.4-0 (setts), ffprobe 5.1.4-0
[debug] Optional libraries: brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, requests-2.28.1, secretstorage-3.3.3, sqlite3-3.40.1, urllib3-1.26.12
[debug] Proxy map: {}
[debug] Request Handlers: urllib
[debug] Loaded 1798 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2023.12.30 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2023.12.30 from yt-dlp/yt-dlp)
[FranceTVSite] Extracting URL: https://www.france.tv/france-2/laissez-vous-guider/laissez-vous-guider-saison-1/5706441-les-merveilles-de-l-egypte-antique.html
[FranceTVSite] 5706441-les-merveilles-de-l-egypte-antique: Downloading webpage
[FranceTV] Extracting URL: francetv:906e5b03-5cb8-4d65-9924-dc60f13f7c5b
[FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: Downloading desktop video JSON
WARNING: [FranceTV] Unable to download JSON metadata: HTTP Error 400: Bad Request
[FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: Downloading mobile video JSON
WARNING: [FranceTV] Unable to download JSON metadata: HTTP Error 400: Bad Request
[debug] Extractor gave empty title. Creating a generic title
ERROR: [FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
Traceback (most recent call last):
  File "/home/laurent/.local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1587, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/laurent/.local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1743, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/laurent/.local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1802, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/laurent/.local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 2781, in process_video_result
    self.raise_no_formats(info_dict)
  File "/home/laurent/.local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1085, in raise_no_formats
    raise ExtractorError(msg, video_id=info['id'], ie=info['extractor'],
yt_dlp.utils.ExtractorError: [FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```

**Comment by pcjco** [comment]
Stopped working very recently (was working 1 day ago). Also when using the nighly build.

Latest version: nightly@2024.02.29.232658 from yt-dlp/yt-dlp-nightly-builds

**Comment by RPisano26** [comment]
I also confirm. Worked like a charm last week (2024-02-23). Same symptom, same error, same log. Last version. On 2024-03-01 doesn't work anymore.

**Comment by kervala** [comment]
I also confirm this issue :)

**Comment by Gigoince** [comment]
Hi there,

I am not familiar with diff files, how could I test this ?

**Comment by Gigoince** [comment]
@bashonly Thanks,

Nope, does not work, different errors, though

[debug] Command-line config: ['-vU', '-F', 'https://www.france.tv/france-2/laissez-vous-guider/laissez-vous-guider-saison-1/5706441-les-merveilles-de-l-egypte-antique.html']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version bashonly/yt-dlp@francetv build 2024.03.01.142221 [4752ff496] (zip)
[debug] Python 3.11.2 (CPython x86_64 64bit) - Linux-6.1.0-17-amd64-x86_64-with-glibc2.36 (OpenSSL 3.0.11 19 Sep 2023, glibc 2.36)
[debug] exe versions: ffmpeg 5.1.4-0 (setts), ffprobe 5.1.4-0
[debug] Optional libraries: brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, requests-2.28.1, secretstorage-3.3.3, sqlite3-3.40.1, urllib3-1.26.12
[debug] Proxy map: {}
[debug] Request Handlers: urllib
[debug] Loaded 1834 extractors
[debug] Fetching release info: https://api.github.com/repos/bashonly/yt-dlp/releases/tags/francetv
Requested version: bashonly/yt-dlp@francetv build 2024.03.01.142221
yt-dlp is up to date (bashonly/yt-dlp@francetv build 2024.03.01.142221)
[FranceTVSite] Extracting URL: https://www.france.tv/france-2/laissez-vous-guider/laissez-vous-guider-saison-1/5706441-les-merveilles-de-l-egypte-antique.html
[FranceTVSite] 5706441-les-merveilles-de-l-egypte-antique: Downloading webpage
[FranceTV] Extracting URL: francetv:906e5b03-5cb8-4d65-9924-dc60f13f7c5b#__youtubedl_smuggle=%7B%22origin%22%3A+%22www.france.tv%22%7D
[FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: Downloading desktop video JSON
[FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: Downloading mobile video JSON
[FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: Downloading signed dash manifest URL
[FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: Downloading MPD manifest
WARNING: [FranceTV] Failed to download MPD manifest: HTTP Error 403: Forbidden
[FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: Downloading signed dash manifest URL
[FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: Downloading MPD manifest
WARNING: [FranceTV] Failed to download MPD manifest: HTTP Error 403: Forbidden
ERROR: [FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
Traceback (most recent call last):
  File "/home/laurent/.local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1593, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/laurent/.local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1749, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/laurent/.local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1808, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/laurent/.local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 2795, in process_video_result
    self.raise_no_formats(info_dict)
  File "/home/laurent/.local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1091, in raise_no_formats
    raise ExtractorError(msg, video_id=info['id'], ie=info['extractor'],
yt_dlp.utils.ExtractorError: [FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U

**Comment by RPisano26** [comment]
Same thing, another error:
[FranceTVSite] Extracting URL: https://www.france.tv/france-3/wakfu/saison-4/5754204-toross-mordal.html
[FranceTVSite] 5754204-toross-mordal: Downloading webpage
[FranceTV] Extracting URL: francetv:c2ba3743-ceb2-40e0-b68d-2e57fa2bae10#__youtubedl_smuggle=%7B%22origin%22%3A+%22www.france.tv%22%7D
[FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: Downloading desktop video JSON
[FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: Downloading mobile video JSON
[FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: Downloading signed dash manifest URL
[FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: Downloading MPD manifest
WARNING: [FranceTV] Failed to download MPD manifest: HTTP Error 403: Forbidden
[FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: Downloading signed dash manifest URL
[FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: Downloading MPD manifest
WARNING: [FranceTV] Failed to download MPD manifest: HTTP Error 403: Forbidden
ERROR: [FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U

D:\dvdrip>yt-dlp -vU https://www.france.tv/france-3/wakfu/saison-4/5754204-toross-mordal.html
[debug] Command-line config: ['-vU', 'https://www.france.tv/france-3/wakfu/saison-4/5754204-toross-mordal.html']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version bashonly/yt-dlp@francetv build 2024.03.01.142221 [4752ff496] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 2023-11-15-git-78f55457c9-full_build-www.gyan.dev (setts), ffprobe 2023-11-15-git-78f55457c9-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.02.02, mutagen-1.47.0, requests-2.31.0, sqlite3-3.35.5, urllib3-2.2.1, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1834 extractors
[debug] Fetching release info: https://api.github.com/repos/bashonly/yt-dlp/releases/tags/francetv
Requested version: bashonly/yt-dlp@francetv build 2024.03.01.142221
yt-dlp is up to date (bashonly/yt-dlp@francetv build 2024.03.01.142221)
[FranceTVSite] Extracting URL: https://www.france.tv/france-3/wakfu/saison-4/5754204-toross-mordal.html
[FranceTVSite] 5754204-toross-mordal: Downloading webpage
[FranceTV] Extracting URL: francetv:c2ba3743-ceb2-40e0-b68d-2e57fa2bae10#__youtubedl_smuggle=%7B%22origin%22%3A+%22www.france.tv%22%7D
[FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: Downloading desktop video JSON
[FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: Downloading mobile video JSON
[FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: Downloading signed dash manifest URL
[FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: Downloading MPD manifest
WARNING: [FranceTV] Failed to download MPD manifest: HTTP Error 403: Forbidden
[FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: Downloading signed dash manifest URL
[FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: Downloading MPD manifest
WARNING: [FranceTV] Failed to download MPD manifest: HTTP Error 403: Forbidden
ERROR: [FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1593, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1749, in __extract_info
  File "yt_dlp\YoutubeDL.py", line 1808, in process_ie_result
  File "yt_dlp\YoutubeDL.py", line 2795, in process_video_result
  File "yt_dlp\YoutubeDL.py", line 1091, in raise_no_formats
yt_dlp.utils.ExtractorError: [FranceTV] c2ba3743-ceb2-40e0-b68d-2e57fa2bae10: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U

**Comment by bashonly** [comment]
Does the site make you log in to watch these videos?

**Comment by Gigoince** [comment]
Yes it does.

**Comment by RPisano26** [comment]
using --username and --password does not work either.

**Comment by bashonly** [comment]
Might not help either, but have you tried passing cookies to yt-dlp?

**Comment by Gigoince** [comment]
Passing cookies does not help (if I did it right !) Here is the command I entered 

`yt-dlp --cookies-from-browser firefox:fbfimf2h.Dev https://www.france.tv/france-2/laissez-vous-guider/laissez-vous-guider-saison-1/5706441-les-merveilles-de-l-egypte-antique.html`

And here is the result :

[FranceTVSite] Extracting URL: https://www.france.tv/france-2/laissez-vous-guider/laissez-vous-guider-saison-1/5706441-les-merve...-egypte-antique.html
[FranceTVSite] 5706441-les-merveilles-de-l-egypte-antique: Downloading webpage
Extracting cookies from firefox
Extracted 57 cookies from firefox
[FranceTV] Extracting URL: francetv:906e5b03-5cb8-4d65-9924-dc60f13f7c5b#__youtubedl_smuggle=%7B%22origin%22%3A+%22www.france.tv%22%7D
[FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: Downloading desktop video JSON
[FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: Downloading mobile video JSON
[FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: Downloading signed dash manifest URL
[FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: Downloading MPD manifest
WARNING: [FranceTV] Failed to download MPD manifest: HTTP Error 403: Forbidden
[FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: Downloading signed dash manifest URL
[FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: Downloading MPD manifest
WARNING: [FranceTV] Failed to download MPD manifest: HTTP Error 403: Forbidden
ERROR: [FranceTV] 906e5b03-5cb8-4d65-9924-dc60f13f7c5b: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U

**Comment by JulienPalard** [comment]
It stopped working for me on 2024-02-29 between 18h45 CET and 19h45 CET.

The command I was using was:

    yt-dlp "-S" "height:480" "-o" "5714823-emission-du-jeudi-29-fevrier-2024.%(ext)s" "https://www.france.tv/france-5/c-dans-l-air/5714823-emission-du-jeudi-29-fevrier-2024.html"

Before 2024-02-29 it was working without cookies, without username, and without password, just with the previous command (but with the videos of the previous days).

**Comment by bashonly** [comment]
The `account-needed` label was added to this issue because an account will most likely be needed by a dev in order to fix this. And if/when it is fixed, then the extractor may require cookies/credentials as well

**Comment by pcjco** [comment]
No account is needed to watch video on FranceTV portal. I am able to watch videos from browser as long as i am geo-localized in France. Not sure if this apply to ALL videos though but this one is working without account: [link](https://www.france.tv/france-5/bluey/saison-2/4259296-le-gecko-collant.html)

**Comment by Gigoince** [comment]
@pcjco Most videos ask me for credentials (I am located in Brittany and am not using any VPN)

**Comment by bashonly** [comment]
new patch for some in-region to test with:
```diff
diff --git a/yt_dlp/extractor/francetv.py b/yt_dlp/extractor/francetv.py
index 0ceecde74..bc9fa9fc0 100644
--- a/yt_dlp/extractor/francetv.py
+++ b/yt_dlp/extractor/francetv.py
@@ -1,21 +1,29 @@
+import urllib.parse
+
 from .common import InfoExtractor
 from .dailymotion import DailymotionIE
 from ..utils import (
     ExtractorError,
     determine_ext,
+    filter_dict,
     format_field,
     int_or_none,
     join_nonempty,
     parse_iso8601,
     parse_qs,
+    smuggle_url,
+    unsmuggle_url,
+    url_or_none,
 )
 
 
 class FranceTVBaseInfoExtractor(InfoExtractor):
-    def _make_url_result(self, video_or_full_id, catalog=None):
+    def _make_url_result(self, video_or_full_id, catalog=None, url=None):
         full_id = 'francetv:%s' % video_or_full_id
         if '@' not in video_or_full_id and catalog:
             full_id += '@%s' % catalog
+        if url:
+            full_id = smuggle_url(full_id, {'origin': urllib.parse.urlparse(url).hostname})
         return self.url_result(
             full_id, ie=FranceTVIE.ie_key(),
             video_id=video_or_full_id.split('@')[0])
@@ -76,7 +84,7 @@ class FranceTVIE(InfoExtractor):
         'only_matching': True,
     }]
 
-    def _extract_video(self, video_id, catalogue=None):
+    def _extract_video(self, video_id, catalogue=None, origin=None):
         # Videos are identified by idDiffusion so catalogue part is optional.
         # However when provided, some extra formats may be returned so we pass
         # it if available.
@@ -94,10 +102,11 @@ def _extract_video(self, video_id, catalogue=None):
         for device_type in ('desktop', 'mobile'):
             dinfo = self._download_json(
                 'https://player.webservices.francetelevisions.fr/v1/videos/%s' % video_id,
-                video_id, 'Downloading %s video JSON' % device_type, query={
+                video_id, 'Downloading %s video JSON' % device_type, query=filter_dict({
                     'device_type': device_type,
                     'browser': 'chrome',
-                }, fatal=False)
+                    'domain': origin,
+                }), fatal=False)
 
             if not dinfo:
                 continue
@@ -130,18 +139,17 @@ def _extract_video(self, video_id, catalogue=None):
         subtitles = {}
         for video in videos:
             format_id = video.get('format')
+            video_url = url_or_none(video.get('url'))
+            if not video_url:
+                continue
 
-            video_url = None
             if video.get('workflow') == 'token-akamai':
-                token_url = video.get('token')
-                if token_url:
+                if token_url := url_or_none(video.get('token')):
                     token_json = self._download_json(
-                        token_url, video_id,
-                        'Downloading signed %s manifest URL' % format_id)
-                    if token_json:
-                        video_url = token_json.get('url')
-            if not video_url:
-                video_url = video.get('url')
+                        token_url, video_id, f'Downloading signed {format_id} manifest URL',
+                        fatal=False, query={'format': 'json', 'url': video_url}) or {}
+                    if tokenized_url := url_or_none(token_json.get('url')):
+                        video_url = tokenized_url
 
             ext = determine_ext(video_url)
             if ext == 'f4m':
@@ -213,6 +221,7 @@ def _extract_video(self, video_id, catalogue=None):
         }
 
     def _real_extract(self, url):
+        url, smuggled_data = unsmuggle_url(url, {})
         mobj = self._match_valid_url(url)
         video_id = mobj.group('id')
         catalog = mobj.group('catalog')
@@ -224,7 +233,7 @@ def _real_extract(self, url):
             if not video_id:
                 raise ExtractorError('Invalid URL', expected=True)
 
-        return self._extract_video(video_id, catalog)
+        return self._extract_video(video_id, catalog, origin=smuggled_data.get('origin'))
 
 
 class FranceTVSiteIE(FranceTVBaseInfoExtractor):
@@ -314,7 +323,7 @@ def _real_extract(self, url):
                 r'(?:href=|player\.setVideo\(\s*)"http://videos?\.francetv\.fr/video/([^@]+@[^"]+)"',
                 webpage, 'video ID').split('@')
 
-        return self._make_url_result(video_id, catalogue)
+        return self._make_url_result(video_id, catalogue, url=url)
 
 
 class FranceTVInfoIE(FranceTVBaseInfoExtractor):
@@ -405,4 +414,4 @@ def _real_extract(self, url):
              r'(?:data-id|<figure[^<]+\bid)=["\']([\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12})'),
             webpage, 'video id')
 
-        return self._make_url_result(video_id)
+        return self._make_url_result(video_id, url=url)
```

I made a new [pre-release](https://github.com/bashonly/yt-dlp/releases/tag/francetv) for testing, which exe/zip users can update to with this command:
```shell
yt-dlp --update-to bashonly/yt-dlp@francetv
```
(if you already tried my last build you need to run this again)

**Comment by Gigoince** [comment]
It does work ! \o/

Thanks a million !

**Comment by Gigoince** [comment]
@bashonly  I was wondering, is there a way to go back to nightly executable and keeping the changes you made ?

**Comment by bashonly** [comment]
@Gigoince No. But that pre-release build is equivalent to the latest nightly release with the francetv patch applied on top. The day after #9333 is merged into master you would be able to `--update-to nightly`

**Comment by Gigoince** [comment]
@bashonly  OK, Great ! Thanks a lot !

**Comment by gilles76** [comment]
Thank you so much ! 24 hours to commit a patch, that is really reactivity. Work fine.

**Comment by HiTiH-63** [comment]
@bashonly thank you so much, it works very well !

**Comment by bashonly** [comment]
@Gigoince @kervala @HiTiH-63 

Would any of you be able to test #9347 ? It's working fine for me, but I can't test with geo-restricted links

Once again I've made a [pre-release](https://github.com/bashonly/yt-dlp/releases/tag/francetv) for ease of testing, which exe/zip users can update to with this command:
```shell
yt-dlp --update-to bashonly/yt-dlp@francetv
```
(if you already tried my last build you need to run this again)

**Comment by bashonly** [comment]
Example geo-restricted URL: https://www.france.tv/france-5/bluey/saison-2/4259296-le-gecko-collant.html

**Comment by Gigoince** [comment]
It works on the link you provided, though with a strange behavior : loads of lines that go like 
`[hls @ 0x55bae8359700] Opening 'crypto+https://cloudreplay.ftven.fr/ZXhwPTE3MDk1MTA3OTd+YWNsPSUyZip+aG1hYz1hZGY4YmQ5NDkxYWFiYTMwODk4ZTlmZTY2OTk5OGIwMDkwY2Q1ODQxMGU5NzNkNjg4YWNlNjZmMTE2YWQ3N2Y0/0cbe13dad9d56/1351395428-1708776515_france-domtom_TA.ism/1351395428-1708776515_france-domtom_TA-audio_fre=96000-50.aac' for reading
[hls @ 0x55bae8359700] Opening 'crypto+https://cloudreplay.ftven.fr/ZXhwPTE3MDk1MTA3OTd+YWNsPSUyZip+aG1hYz1hZGY4YmQ5NDkxYWFiYTMwODk4ZTlmZTY2OTk5OGIwMDkwY2Q1ODQxMGU5NzNkNjg4YWNlNjZmMTE2YWQ3N2Y0/0cbe13dad9d56/1351395428-1708776515_france-domtom_TA.ism/1351395428-1708776515_france-domtom_TA-audio_fre=96000-51.aac' for reading
[hls @ 0x55bae8359700] Opening 'crypto+https://cloudreplay.ftven.fr/ZXhwPTE3MDk1MTA3OTd+YWNsPSUyZip+aG1hYz1hZGY4YmQ5NDkxYWFiYTMwODk4ZTlmZTY2OTk5OGIwMDkwY2Q1ODQxMGU5NzNkNjg4YWNlNjZmMTE2YWQ3N2Y0/0cbe13dad9d56/1351395428-1708776515_france-domtom_TA.ism/1351395428-1708776515_france-domtom_TA-audio_fre=96000-52.aac' for reading
[hls @ 0x55bae8359700] Opening 'crypto+https://cloudreplay.ftven.fr/ZXhwPTE3MDk1MTA3OTd+YWNsPSUyZip+aG1hYz1hZGY4YmQ5NDkxYWFiYTMwODk4ZTlmZTY2OTk5OGIwMDkwY2Q1ODQxMGU5NzNkNjg4YWNlNjZmMTE2YWQ3N2Y0/0cbe13dad9d56/1351395428-1708776515_france-domtom_TA.ism/1351395428-1708776515_france-domtom_TA-audio_fre=96000-53.aac' for reading
`

But on other links, it just does not work at all
` yt-dlp https://www.france.tv/documentaires/societe/l-assiette-francaise/
[generic] Extracting URL: https://www.france.tv/documentaires/societe/l-assiette-francaise/
[generic] l-assiette-francaise: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] l-assiette-francaise: Extracting information
ERROR: Unsupported URL: https://www.france.tv/documentaires/societe/l-assiette-francaise/
`

**Comment by Gigoince** [comment]
yt-dlp -vU https://www.france.tv/documentaires/societe/l-assiette-francaise/
[debug] Command-line config: ['-vU', 'https://www.france.tv/documentaires/societe/l-assiette-francaise/']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version bashonly/yt-dlp@francetv build 2024.03.03.174708 [5e06446e3] (zip)
[debug] Python 3.11.2 (CPython x86_64 64bit) - Linux-6.1.0-17-amd64-x86_64-with-glibc2.36 (OpenSSL 3.0.11 19 Sep 2023, glibc 2.36)
[debug] exe versions: ffmpeg 5.1.4-0 (setts), ffprobe 5.1.4-0
[debug] Optional libraries: brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, requests-2.28.1, secretstorage-3.3.3, sqlite3-3.40.1, urllib3-1.26.12
[debug] Proxy map: {}
[debug] Request Handlers: urllib
[debug] Loaded 1834 extractors
[debug] Fetching release info: https://api.github.com/repos/bashonly/yt-dlp/releases/tags/francetv
Requested version: bashonly/yt-dlp@francetv build 2024.03.03.174708
yt-dlp is up to date (bashonly/yt-dlp@francetv build 2024.03.03.174708)
[generic] Extracting URL: https://www.france.tv/documentaires/societe/l-assiette-francaise/
[generic] l-assiette-francaise: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] l-assiette-francaise: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://www.france.tv/documentaires/societe/l-assiette-francaise/
Traceback (most recent call last):
  File "/home/laurent/.local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1593, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/laurent/.local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1728, in __extract_info
    ie_result = ie.extract(url)
                ^^^^^^^^^^^^^^^
  File "/home/laurent/.local/bin/yt-dlp/yt_dlp/extractor/common.py", line 732, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/laurent/.local/bin/yt-dlp/yt_dlp/extractor/generic.py", line 2531, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.france.tv/documentaires/societe/l-assiette-francaise/

**Comment by bashonly** [comment]
`https://www.france.tv/documentaires/societe/l-assiette-francaise/` doesn't work with master/stable either

**Comment by bashonly** [comment]
> It works on the link you provided, though with a strange behavior : loads of lines that go like

You need to install pycryptodomex

**Comment by Gigoince** [comment]
> `https://www.france.tv/documentaires/societe/l-assiette-francaise/` doesn't work with master/stable either

Indeed ! :-(

**Comment by bashonly** [comment]
The extractor has never supported series URLs. You need to pass the individual episode/part URLs, e.g. `https://www.france.tv/france-2/13h15-le-dimanche/5766081-l-assiette-episode-2-partie-3-au-four-et-au-moulin.html`

**Comment by Gigoince** [comment]
#9347  seems to work fine. (Tested on several links, including the one provided)

I had not realised that the link I had tried was a series, all apologies.

**Comment by bashonly** [comment]
@Gigoince all good, thank you for testing.

**Comment by pokemaster974** [comment]
> @Gigoince all good, thank you for testing.

Hi @bashonly is your last commit to retrieve back the hls stream ?
I've remark that since the fix for the first bug, it displays twice the dash streams :

```
ID                     EXT RESOLUTION FPS |   FILESIZE   TBR PROTO | VCODEC        VBR ACODEC     ABR ASR MORE INFO
-------------------------------------------------------------------------------------------------------------------------------------------------------
dash-audio_qad=96000-0 m4a audio only     | ~ 64.75MiB   96k dash  | audio only        mp4a.40.2  96k 48k [qad] audio description, DASH audio, m4a_dash
dash-audio_qad=96000-1 m4a audio only     | ~ 64.75MiB   96k dash  | audio only        mp4a.40.2  96k 48k [qad] audio description, DASH audio, m4a_dash
dash-audio_fre=96000-0 m4a audio only     | ~ 64.75MiB   96k dash  | audio only        mp4a.40.2  96k 48k [fr] DASH audio, m4a_dash
dash-audio_fre=96000-1 m4a audio only     | ~ 64.75MiB   96k dash  | audio only        mp4a.40.2  96k 48k [fr] DASH audio, m4a_dash
dash-video=400000-0    mp4 384x216     25 | ~269.78MiB  400k dash  | avc1.42C01E  400k video only         DASH video, mp4_dash
dash-video=400000-1    mp4 384x216     25 | ~269.78MiB  400k dash  | avc1.42C01E  400k video only         DASH video, mp4_dash
dash-video=950000-0    mp4 640x360     25 | ~640.72MiB  950k dash  | avc1.4D401F  950k video only         DASH video, mp4_dash
dash-video=950000-1    mp4 640x360     25 | ~640.72MiB  950k dash  | avc1.4D401F  950k video only         DASH video, mp4_dash
dash-video=1400000-0   mp4 960x540     25 | ~944.21MiB 1400k dash  | avc1.4D401F 1400k video only         DASH video, mp4_dash
dash-video=1400000-1   mp4 960x540     25 | ~944.21MiB 1400k dash  | avc1.4D401F 1400k video only         DASH video, mp4_dash
dash-video=2000000-0   mp4 1280x720    25 | ~  1.32GiB 2000k dash  | avc1.64001F 2000k video only         DASH video, mp4_dash
dash-video=2000000-1   mp4 1280x720    25 | ~  1.32GiB 2000k dash  | avc1.64001F 2000k video only         DASH video, mp4_dash
dash-video=5000000-0   mp4 1920x1080   25 | ~  3.29GiB 5000k dash  | avc1.640029 5000k video only         DASH video, mp4_dash
dash-video=5000000-1   mp4 1920x1080   25 | ~  3.29GiB 5000k dash  | avc1.640029 5000k video only         DASH video, mp4_dash
```

I wanted to open an issue but I see that #9347 speaks about m3u8 so if it supposed to fix it, I won't open it.
But I've observed a bug with subtitle (the file seems to be empty) but it probably needs another issue as it not the same bug as here.

**Comment by bashonly** [comment]
@pokemaster974 yes, run `yt-dlp --update-to bashonly/yt-dlp@francetv` to try out #9347

**Comment by NastyPhoenix** [comment]
Thanks a lot ; it works like a charm (yt-dlp --update-to bashonly/yt-dlp@francetv)

**Comment by guy-teube** [comment]
I've just updated to nightly version (thanks for the fix!) but I still have issues with France TV videos:
`
$ yt-dlp -F https://www.france.tv/france-5/t-choupi-a-la-campagne/saison-1/5769363-le-marche-comme-les-grands.html
[FranceTVSite] Extracting URL: https://www.france.tv/france-5/t-choupi-a-la-campagne/saison-1/5769363-le-marche-comme-les-grands.html
[FranceTVSite] 5769363-le-marche-comme-les-grands: Downloading webpage
[FranceTV] Extracting URL: francetv:9f4d3846-19f4-4a8f-aa36-6e8b817a4841#__youtubedl_smuggle=%7B%22hostname%22%3A+%22www.france.tv%22%7D
[FranceTV] 9f4d3846-19f4-4a8f-aa36-6e8b817a4841: Downloading desktop chrome video JSON
[FranceTV] 9f4d3846-19f4-4a8f-aa36-6e8b817a4841: Downloading mobile safari video JSON
WARNING: [FranceTV] Unable to download JSON metadata: HTTP Error 422: Unprocessable Entity
[FranceTV] 9f4d3846-19f4-4a8f-aa36-6e8b817a4841: Downloading MPD manifest
WARNING: This video is drm protected and only images are available for download. use --list-formats to see them
[info] Available formats for 9f4d3846-19f4-4a8f-aa36-6e8b817a4841:
ID           EXT   RESOLUTION │ PROTO │ VCODEC MORE INFO
─────────────────────────────────────────────────────────
spritesheets mhtml unknown    │ mhtml │ images storyboard
`

Jsut before to do the update, I get the "DRM protected" error. So I don't know if the 422 error is on FTV side, or yt-dlp's.

**Comment by bashonly** [comment]
@guy-teube Nope, the video is DRM protected. The 422 error is indicating that there is no HLS available for this video, only DRM-protected DASH.

**Comment by errwann** [comment]
Note that since the same date (29/02/2024), the direct streams of france.tv don't work anymore :
- https://www.france.tv/france-2/direct.html
- https://www.france.tv/france-3/direct.html
- https://www.france.tv/france-4/direct.html
- https://www.france.tv/france-5/direct.html

For example, "yt-dlp https://www.france.tv/france-2/direct.html" now returns :

>   WARNING: [FranceTV] Failed to download m3u8 information: HTTP Error 403: Forbidden
>   ERROR: [FranceTV] 29bdf749-7082-4426-a4f3-595cc436aa0d: No video formats found!

**Comment by diligiant** [comment]
The same problem is back (running 240527). Should I create a new issue?

**Comment by seproDev** [comment]
@diligiant Update to nightly. If you are still running in to problem you can create a new issue.

**Comment by pokemaster974** [comment]
> The same problem is back (running 240527). Should I create a new issue?

I think so. I've seend that yesterday they have change a little thing into endpoint `https://player.webservices.francetelevisions.fr/v1/videos/`
Don't know exactly the method for yt-dlp but it maybe needs a fix.

**Comment by diligiant** [comment]
> @diligiant Update to nightly. If you are still running in to problems you can create a new issue.

thank you. works.

**Comment by scsskid** [comment]
nightly works thanks, great work all
